src/test/java/org/itsallcode/openfasttrace/exporter/specobject/TestSpecobjectExportImport.java

```diff
@@ -0,0 +1,138 @@
+                .id(SpecificationItemId.createId("foo", "bar", 1)) //
+                .description("the description") //
+                .location(Location.create("dummy.xml", 4)) //
+                .build();
+        assertExportAndImport(item);
+    }
+
+    @Test
+    void testExportImportSpecObjectWithOptionalElements() throws IOException, XMLStreamException
+    {
+        final SpecificationItem item = SpecificationItem.builder() //
+                .id(SpecificationItemId.createId("req", "me", 2)) //
+                .title("My item title") //
+                .status(ItemStatus.DRAFT) //
+                .description("the description") //
+                .rationale("the rationale") //
+                .comment("the comment") //
+                .addCoveredId("feat", "covered", 1) //
+                .addDependOnId("req", "depend-on", 1) //
+                .addNeedsArtifactType("impl") //
+                .addTag("the tag") //
+                .location(Location.create("dummy.xml", 4)) //
+                .build();
+        assertExportAndImport(item);
+    }
+
+    @Test
+    void testExportImportTwoSpecObjects() throws IOException, XMLStreamException
+    {
+        final SpecificationItem itemA = SpecificationItem.builder() //
+                .id(SpecificationItemId.createId("foo", "bar", 1)) //
+                .status(ItemStatus.PROPOSED) //
+                .description("the description") //
+                .rationale("the rationale") //
+                .comment("the comment") //
+                .location(Location.create("dummy.xml", 4)) //
+                .build();
+        final SpecificationItem itemB = SpecificationItem.builder() //
+                .id(SpecificationItemId.createId("baz", "zoo", 2)) //
+                .status(ItemStatus.REJECTED) //
+                .description("another\ndescription") //
+                .rationale("another\nrationale") //
+                .comment("another\ncomment") //
+                .location(Location.create("dummy.xml", 5)) //
+                .build();
+        assertExportAndImport(itemA, itemB);
+    }
+
+    private void assertExportAndImport(final SpecificationItem... items)
+    {
+        final String exportedItems = exportToString(items);
+        final List<SpecificationItem> importedItems = importItems(exportedItems);
+        System.out.println(importedItems);
+        System.out.println(asList(items));
+
+        assertThat(importedItems, AutoMatcher.contains(items));
+    }
+
+    private List<SpecificationItem> importItems(final String exportedItems)
+    {
+        final InputFile input = StreamInput.forContent(Paths.get("dummy.xml"), exportedItems);
+        final SpecificationListBuilder itemBuilder = SpecificationListBuilder.create();
+        new SpecobjectImporterFactory().createImporter(input, itemBuilder).runImport();
+        return itemBuilder.build();
+    }
+
+    private String exportToString(final SpecificationItem... items)
```

Comment:
Alternatively use the System Out Extension to make test more compact and remove potential duplication with other tests.

Reply:
In my opinion extensions like this should only be used if there is no other solution.